src/cmd-tag

```diff
@@ -0,0 +1,177 @@
+                        required=True)
+    delete.set_defaults(func=cmd_delete)
+
+    list_tags = subparsers.add_parser("list", help="List available tags in "
+                                                   "the metadata")
+    list_tags.set_defaults(func=cmd_list)
+
+    update = subparsers.add_parser("update", help="Update existing tag or "
+                                                  "create a new tag")
+    update.add_argument("--tag", help="Tag to be updated", required=True)
+    update.add_argument("--build", help="Build to update tag with",
+                        required=True)
+    update.add_argument("--force", help="Force the update of a tag",
+                        action="store_true")
+    update.set_defaults(func=cmd_update)
+
+    return parser.parse_args()
+
+
+def init_build_data():
+    """Initialize build metadata"""
+
+    builds_json = "builds/builds.json"
+
+    if os.path.isfile(builds_json):
+        with open(builds_json) as json_file:
+            build_data = json.load(json_file)
+    else:
+        print("INFO: Did not find existing builds.json; starting with empty "
+              "build list and tags list")
+        build_data = {"builds": [], "tags": []}
+
+    return build_data
+
+
+def finalize_json(build_data):
+    """Wrapper around cmdlib.write_json()"""
+
+    builds_json = "builds/builds.json"
+
+    build_data['timestamp'] = rfc3339_time()
+    write_json(builds_json, build_data)
+
+
+def available_builds(build_data):
+    """Returns the availble build IDs from build metadata"""
+
+    return build_data["builds"]
+
+
+def available_tags(build_data):
+    """Returns the available tag names from build metadata"""
+
+    return [t.get("name") for t in build_data.get("tags", {})]
+
+
+def cmd_list(args):
+    """List available tags in build metadata"""
+
+    build_data = init_build_data()
+    avail_tags = available_tags(build_data)
+    if not avail_tags:
+        print("No tags found")
+        return
+
+    for tag in build_data["tags"]:
+        print(json.dumps({"name": tag.get("name"),
+                          "created": tag.get("created"),
+                          "target": tag.get("target")}))
+
+
+def cmd_update(args):
+    """Create or update a tag to new build ID"""
+
+    build_data = init_build_data()
+    avail_tags = available_tags(build_data)
+    avail_builds = available_builds(build_data)
+
+    if "tags" not in build_data:
+        build_data["tags"] = []
+
+    # tag not available, so let's create it
+    if args.tag not in avail_tags:
+        # build not available, check for force
+        if args.build not in avail_builds:
+            # force not used, gotta bail
+            if not args.force:
+                fatal("Cannot create a tag with a build that does not exist")
+            if args.force:
+                print(f"INFO: Creating a new tag with a build ({args.build}) "
+                      "that does not exist")
```

Comment:
Looks like f-strings are [new in 3.6][1].  My RHEL 7.5 CSB has 3.4.9.  Is it worth using an f-string here vs. the slightly longer but more widely supported:

```python
print('INFO: Creating a new tag with a build ({}) that does not exist'.format(args.build))
```

?
[1]: https://www.python.org/dev/peps/pep-0498/#raw-f-strings

Reply:
f-strings are used in other parts of the code base and this tool will typically be invoked via the container (which is [built on Fedora 28](https://github.com/coreos/coreos-assembler/blob/master/Dockerfile#L1)), so I feel OK sticking with them.

Reply:
>  My RHEL 7.5 CSB has 3.4.9. 

You shouldn't use the version of `python` that lives in the root partition of your hard drive - use containers!  (`coreos-assembler` is intended to always come as a container or be installed inside one)